02_Development/Display/spl06_007.py

```diff
@@ -0,0 +1,703 @@
+            raise ValueError("Temperature oversampling can only be "
+                             "1, 2, 4, 8, 16, 32, 64, or 128X")
+        self._i2c.write_register(SensorConstants.TEMPERATURE_CONFIGURATION,
+                                 rate_mode | oversample_mode)
+        if oversample > 8:
+            self._i2c.write_register(
+                SensorConstants.INTERRUPT_AND_FIFO_CONFIGURATION,
+                SensorConstants.TEMPERATURE_RESULT_BIT_SHIFT
+            )
+        else:
+            new_interrupt_and_fifo_config_state = (
+                self._i2c.read_register(
+                    SensorConstants.INTERRUPT_AND_FIFO_CONFIGURATION)
+            ) & (0xff - SensorConstants.TEMPERATURE_RESULT_BIT_SHIFT)
+            self._i2c.write_register(
+                SensorConstants.INTERRUPT_AND_FIFO_CONFIGURATION,
+                new_interrupt_and_fifo_config_state
+            )
+
+    def raw_temperature(self):
+        """The raw temperature reading from the sensor.  This need to
+        be scaled and compensated per the data sheet to be useful.
+        """
+        if self._op_mode == PressureSensor.OpMode.command:
+            self._i2c.write_register(SensorConstants.SENSOR_OP_MODE,
+                                     SensorConstants.COMMAND_TEMPERATURE)
+
+        while (self._i2c.read_register(SensorConstants.SENSOR_OP_MODE)
+               & SensorConstants.TMP_RDY) == 0:
+            time.sleep(self._READY_WAIT_TIME)
+
+        temperature_msb = self._i2c.read_register(
+            SensorConstants.TEMPERATURE_MSB)
+        temperature_lsb = self._i2c.read_register(
+            SensorConstants.TEMPERATURE_LSB)
+        temperature_xlsb = self._i2c.read_register(
+            SensorConstants.TEMPERATURE_XLSB)
+        temperature = self._twos_complement(((temperature_msb << 16)
+                                             + (temperature_lsb << 8)
+                                             + temperature_xlsb),
+                                            24)
+        return temperature
+
+    @property
+    def temperature_scale_factor(self):
+        """A constant used for calibrating the sensor."""
+        return self._temperature_scale_factor
+
+    def _calculate_calibration_coefficients(self):
+        """The SPL06-007 is a calibrated sensor and contains
+        calibration coefficients. These are used in the
+        application (for instance by the host processor) to
+        compensate the measurement results for sensor nonlinearity’s.
+
+        Note: The coefficients read from the coefficient register
+        {c00, c10, c01, c11, c20, c21, c30} are 16 bit 2´s complement
+        numbers.
+
+        Note: The coefficients read from the coefficient register
+        {c0, c1} 12 bit 2´s complement numbers.
+        """
+
+        while (self._i2c.read_register(SensorConstants.SENSOR_OP_MODE)
+               & SensorConstants.COEF_RDY) == 0:
+            time.sleep(self._READY_WAIT_TIME)
```

Comment:
Can you get stuck in an infinite loop here?

Reply:
Yes you can I need to add a timeout